ludwig/schema/utils.py

```diff
@@ -260,10 +260,11 @@ def RegularizerOptions(
 def String(
     description: str,
     default: Union[None, str] = None,
-    allow_none: bool = True,
+    allow_none: bool = False,
     pattern: str = None,
     parameter_metadata: ParameterMetadata = None,
 ):
+    allow_none = allow_none or default is None
```

Comment:
This is too implicit. I prefer making it explcit on the caller side. If `allow_none` is `False` they must choose a non-null default.